--- a/setup.py	2021-10-13 15:04:56.840925521 -0400
+++ b/setup.py	2021-10-14 16:15:24.934360987 -0400
@@ -21,7 +21,7 @@
     url="https://github.com/SpikeInterface/spikeextractors",
     long_description=long_description,
     long_description_content_type="text/markdown",
-    packages=setuptools.find_packages(),
+    packages=setuptools.find_packages(exclude=["tests"]),
     package_data={},
     include_package_data=True,
     install_requires=[
