--- a/setup.py    2018-06-26 04:20:04.000000000 +0200
+++ b/setup.py    2018-11-06 17:47:28.940450375 +0100
@@ -105,6 +105,7 @@
     """
     user_options = [
         ('skip-build', None, 'skip building the C/C++ code (assumes it has already been done)'),
+        ('parallel=', 'j', "number of parallel build jobs")
         ]
     boolean_options = ['skip-build']
 
@@ -112,6 +113,7 @@
     def initialize_options(self):
         orig_build.initialize_options(self)
         self.skip_build = '--skip-build' in sys.argv
+        self.parallel = None
 
     def finalize_options(self):
         orig_build.finalize_options(self)
