--- a/wscript   2018-06-26 07:20:04.000000000 +0500
+++ b/wscript	2018-10-29 20:10:06.236956371 +0500
@@ -131,8 +131,8 @@
         _copyEnvGroup(conf.env, '_WX', '_WXGL')
         conf.env.LIB_WXGL += cfg.makeLibName('gl')
 
-        _copyEnvGroup(conf.env, '_WX', '_WXWEBVIEW')
-        conf.env.LIB_WXWEBVIEW += cfg.makeLibName('webview')
+        #_copyEnvGroup(conf.env, '_WX', '_WXWEBVIEW')
+        #conf.env.LIB_WXWEBVIEW += cfg.makeLibName('webview')
 
         _copyEnvGroup(conf.env, '_WX', '_WXXML')
         conf.env.LIB_WXXML += cfg.makeLibName('xml', isMSWBase=True)
@@ -220,16 +220,16 @@
                        uselib_store='WXGL', mandatory=True,
                        msg='Finding libs for WXGL')
 
-        conf.check_cfg(path=conf.options.wx_config, package='',
-                       args='--cxxflags --libs webview,core,net' + rpath,
-                       uselib_store='WXWEBVIEW', mandatory=True,
-                       msg='Finding libs for WXWEBVIEW')
-
-        if isDarwin:
-            conf.check_cfg(path=conf.options.wx_config, package='',
-                           args='--cxxflags --libs core,net' + rpath,
-                           uselib_store='WXWEBKIT', mandatory=True,
-                           msg='Finding libs for WXWEBKIT')
+        #conf.check_cfg(path=conf.options.wx_config, package='',
+        #               args='--cxxflags --libs webview,core,net' + rpath,
+        #               uselib_store='WXWEBVIEW', mandatory=True,
+        #               msg='Finding libs for WXWEBVIEW')
+
+        #if isDarwin:
+        #    conf.check_cfg(path=conf.options.wx_config, package='',
+        #                   args='--cxxflags --libs core,net' + rpath,
+        #                   uselib_store='WXWEBKIT', mandatory=True,
+        #                   msg='Finding libs for WXWEBKIT')
 
         conf.check_cfg(path=conf.options.wx_config, package='',
                        args='--cxxflags --libs xml,core,net' + rpath,
@@ -567,7 +567,7 @@
     makeETGRule(bld, 'etg/_stc.py',        '_stc',       'WXSTC')
     makeETGRule(bld, 'etg/_html.py',       '_html',      'WXHTML')
     makeETGRule(bld, 'etg/_glcanvas.py',   '_glcanvas',  'WXGL')
-    makeETGRule(bld, 'etg/_html2.py',      '_html2',     'WXWEBVIEW')
+    #makeETGRule(bld, 'etg/_html2.py',      '_html2',     'WXWEBVIEW')
     makeETGRule(bld, 'etg/_xml.py',        '_xml',       'WXXML')
     makeETGRule(bld, 'etg/_xrc.py',        '_xrc',       'WXXRC')
     makeETGRule(bld, 'etg/_richtext.py',   '_richtext',  'WXHTML WXRICHTEXT')
@@ -577,8 +577,8 @@
     makeETGRule(bld, 'etg/_aui.py',        '_aui',       'WXAUI')
 
     # Modules that are platform-specific
-    if isDarwin:
-        makeETGRule(bld, 'etg/_webkit.py', '_webkit',    'WXWEBKIT')
+    #if isDarwin:
+        #makeETGRule(bld, 'etg/_webkit.py', '_webkit',    'WXWEBKIT')
     if isWindows:
         makeETGRule(bld, 'etg/_msw.py',    '_msw',       'WX')
