--- a/src/pubsub/core/topicdefnprovider.py      2013-11-17 12:42:54.000000000 -0600
+++ b/src/pubsub/core/topicdefnprovider.py      2019-02-06 17:15:06.138101941 -0600
@@ -7,10 +7,8 @@
 import os, re, inspect
 from textwrap import TextWrapper, dedent
 
-from .. import (
-    policies, 
-    py2and3
-)
+import policies, py2and3
+
 from .topicargspec import (
     topicArgsFromCallable, 
     ArgSpecGiven
